include/tvm/node/container.h

```diff
@@ -1287,6 +1287,18 @@ class Map : public ObjectRef {
     data_ = other.data_;
     return *this;
   }
+  /*!
+   * \brief Merge with another Map. It does not mutate the current one.
+   * \param other Map to be merged.
+   * @return The merged Array. Original Map is kept unchanged.
+   */
+  Map<K, V> Merge(const Map<K, V>& other) const {
```

Comment:
Shall we make it as a global function instead of member? so it is not ambiguitous (that the result is a new map)